src/components/pages/roster-page/roster-page.tsx

```diff
@@ -142,16 +185,198 @@ export const RosterPage = () => {
     setRowsPerPage(newRowsPerPage);
   };
 
-  const handleAlertClose = () => {
-    setAlert({ open: false, message: '', title: '' });
+  const editButtonClicked = async (rosterEntry: ApiRosterEntry) => {
+    // remove any properties that came back as null before sending it to the dialog
+    Object.keys(rosterEntry).forEach((key: string) => (rosterEntry[key] == null) && delete rosterEntry[key]);
+    setSelectedRosterEntry(rosterEntry);
+    setEditRosterEntryDialogProps({
+      open: true,
+      orgId,
+      rosterColumnInfos,
+      rosterEntry,
+      onClose: async () => {
+        setEditRosterEntryDialogProps({ open: false });
+        setSelectedRosterEntry(undefined);
+        await initializeTable();
+      },
+      onError: (message: string) => {
+        setAlertDialogProps({
+          open: true,
+          title: 'Edit Roster Entry',
+          message: `Unable to edit roster entry: ${message}`,
+          onClose: () => { setAlertDialogProps({ open: false }); },
+        });
+      },
+    });
+  };
+
+  const addButtonClicked = async () => {
+    setEditRosterEntryDialogProps({
+      open: true,
+      orgId,
+      rosterColumnInfos,
+      onClose: async () => {
+        setEditRosterEntryDialogProps({ open: false });
+        setSelectedRosterEntry(undefined);
+        await initializeTable();
+      },
+      onError: (message: string) => {
+        setAlertDialogProps({
+          open: true,
+          title: 'Add Roster Entry',
+          message: `Unable to add roster entry: ${message}`,
+          onClose: () => { setAlertDialogProps({ open: false }); },
+        });
+      },
+    });
+  };
+
+  function deleteButtonClicked(rosterEntry: ApiRosterEntry) {
+    setSelectedRosterEntry(rosterEntry);
+    setDeleteRosterEntryDialogOpen(true);
+  }
+
+  const deleteRosterEntry = async () => {
+    try {
+      setDeleteRosterEntryLoading(true);
+      await axios.delete(`api/roster/${orgId}/${selectedRosterEntry!.edipi}`);
+    } catch (error) {
+      let message = 'Internal Server Error';
+      if (error.response?.data?.errors && error.response.data.errors.length > 0) {
+        message = error.response.data.errors[0].message;
+      }
+      setAlertDialogProps({
+        open: true,
+        title: 'Delete Roster Entry',
+        message: `Unable to delete roster entry: ${message}`,
+        onClose: () => { setAlertDialogProps({ open: false }); },
+      });
+    } finally {
+      setDeleteRosterEntryLoading(false);
+      setDeleteRosterEntryDialogOpen(false);
+      setSelectedRosterEntry(undefined);
+      await initializeTable();
+    }
+  };
+
+  const cancelDeleteRosterEntryDialog = () => {
+    setDeleteRosterEntryDialogOpen(false);
+    setSelectedRosterEntry(undefined);
+  };
+
+  const downloadCSVExport = async () => {
```

Comment:
The only reason for adding those methods was because of that proxy issue I ran into in development builds. I agree though it's cleaner just having those in href tags. Would that be preferred even if they only work in production builds?